src/components/ProductCard.tsx

```diff
@@ -30,56 +32,69 @@ const ProductCard = ({ product, quantity, showDynamicPrice }: ProductProp) => {
   const { savedProducts, setSavedProducts } = useContext(ProductContext);
   const [counterValue, setCounterValue] = useState<number>(0);
   var productContent = product.content;
+  const [isModalVisible, setIsModalVisible] = useState(false);
 
   useEffect(() => {
     setCounterValue(getQuantityByProductId(savedProducts, product.id)!);
   }, [savedProducts]);
 
   return (
-    <View className="flex flex-row justify-between mt-4">
-      <View className="flex flex-row ">
-        <View className="w-32 h-24">
-          <Image
-            style={{
-              width: "100%",
-              height: "100%",
-            }}
-            source={{ uri: productContent.image.filename }}
-          />
-        </View>
-        <View className="flex justify-between my-1 ml-2 ">
-          <View>
-            <Text className="flex font-medium">{productContent.name}</Text>
-            <Text className="text-gray-700">{productContent.unit}</Text>
-            <Text className="text-gray-700">
-              Origin: {productContent.origin}
+    <>
+      <ProductModal
+        product={product}
+        isOpen={isModalVisible}
+        setIsOpen={setIsModalVisible}
+        counterValue={counterValue}
+        setCounterValue={setCounterValue}
+      />
+      <TouchableOpacity
+        onPress={() => setIsModalVisible(true)}
+        className="flex flex-row justify-between mt-4"
+      >
+        <View className="flex flex-row ">
+          <View className="w-32 h-24">
+            <Image
+              style={{
+                width: "100%",
+                height: "100%",
+              }}
+              source={{ uri: productContent.image.filename }}
+            />
+          </View>
+          <View className="flex justify-between my-1 ml-2 ">
+            <View>
+              <Text className="flex font-medium">{productContent.name}</Text>
+              <Text className="text-gray-700">{productContent.unit}</Text>
+              <Text className="text-gray-700">
+                Origin: {productContent.origin}
+              </Text>
+            </View>
+            <Text className="font-bold">
+              {parseCurrency(
```

Comment:
Wouldn't be better to have this logic in a function or variable and then just use it here to make readability easier?